pkg/agent/core/ngt/handler/grpc/index_test.go

```diff
@@ -508,106 +508,529 @@ func Test_server_CreateIndex(t *testing.T) {
 func Test_server_SaveIndex(t *testing.T) {
 	t.Parallel()
 	type args struct {
-		ctx context.Context
 		in1 *payload.Empty
 	}
 	type fields struct {
-		name              string
-		ip                string
-		ngt               service.NGT
-		eg                errgroup.Group
-		streamConcurrency int
+		srvOpts   []Option
+		svcCfg    *config.NGT
+		svcOpts   []service.Option
+		indexPath string // index path for svcOpts
 	}
 	type want struct {
 		wantRes *payload.Empty
-		err     error
+		errCode codes.Code
 	}
 	type test struct {
 		name       string
 		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want, *payload.Empty, error) error
-		beforeFunc func(args)
-		afterFunc  func(args)
+		checkFunc  func(test, context.Context, Server, service.NGT, want, *payload.Empty, error) error
+		beforeFunc func(*testing.T, context.Context, Server, service.NGT)
+		afterFunc  func(test)
 	}
-	defaultCheckFunc := func(w want, gotRes *payload.Empty, err error) error {
-		if !errors.Is(err, w.err) {
-			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+	defaultCheckFunc := func(test test, ctx context.Context, s Server, n service.NGT, w want, gotRes *payload.Empty, err error) error {
+		if (err == nil && w.errCode != 0) || (err != nil && w.errCode == 0) {
+			return errors.Errorf("got error is %v, but want error code is %v", err, w.errCode)
 		}
-		if !reflect.DeepEqual(gotRes, w.wantRes) {
+		if err != nil {
+			st, ok := status.FromError(err)
+			if !ok {
+				return errors.Errorf("got error cannot convert to Status: \"%#v\"", err)
+			}
+			if st.Code() != w.errCode {
+				return errors.Errorf("got code: \"%#v\",\n\t\t\t\twant code: \"%#v\"", st.Code(), w.errCode)
+			}
+		}
+
+		if diff := comparator.Diff(gotRes, w.wantRes, comparator.IgnoreUnexported(payload.Empty{})); diff != "" {
 			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", gotRes, w.wantRes)
 		}
 		return nil
 	}
+
+	// common variables for test
+	const (
+		name = "vald-agent-ngt-1" // agent name
+		dim  = 3                  // vector dimension
+		id   = "uuid-1"           // id for getObject request
+	)
+	var (
+		ip = net.LoadLocalIP() // agent ip address
+
+		// default NGT configuration for test
+		defaultSvcCfg = &config.NGT{
+			Dimension:    dim,
+			DistanceType: ngt.Angle.String(),
+			ObjectType:   ngt.Float.String(),
+			KVSDB:        &config.KVSDB{},
+			VQueue:       &config.VQueue{},
+		}
+		defaultSrvOpts = []Option{
+			WithName(name),
+			WithIP(ip),
+		}
+		defaultSvcOpts = []service.Option{
+			service.WithEnableInMemoryMode(false),
+		}
+
+		emptyPayload = &payload.Empty{}
+	)
+
+	mkdirTemp := func() string {
+		d, err := os.MkdirTemp("", "")
+		if err != nil {
+			t.Error(err)
+		}
+		return d
+	}
+	defaultAfterFunc := func(test test) {
+		os.RemoveAll(test.fields.indexPath)
+	}
+	checkBackupFolder := func(fields fields, ctx context.Context, wantVecs []*payload.Insert_Request) error {
```

Comment:
could you please add the description when it is used.
And, i think it would be better to move it into the `defaultCheckFunc`.

Reply:
Did you add the discription?

Reply:
added :)